src/proxy-ui-api/src/main/resources/openapi-definition.yaml

```diff
@@ -3111,17 +3111,47 @@ paths:
         - security server
       summary: upload a configuration anchor file
       operationId: uploadAnchor
-      description: Administrator uploads configuration anchor file
+      description: Administrator uploads a configuration anchor file
       requestBody:
         description: configuration anchor
         content:
-          application/xml:
+          application/octet-stream:
+            schema:
+              type: string
+              format: binary
+              description: configuration anchor file
+      responses:
+        '201':
+          description: configuration anchor uploaded
+        '400':
+          description: request was invalid
+        '401':
+          description: authentication credentials are missing
+        '403':
+          description: request has been refused
+        '404':
+          description: resource requested does not exists
+        '406':
+          description: request specified an invalid format
+        '409':
+          description: an existing item already exists
+        '500':
+          description: internal server error
+  /system/anchor/preview:
+    post: # ok
+      tags:
+        - security server
+      summary: Read the configuration anchor file and return the hash for a preview. The anchor will not be saved.
```

Comment:
Yes the anchor's instance needs to match the current instance, otherwise an error will be returned. Will add a mention about the validation part. 

Should we improve the API definition to also include this kind of error scenarios in the summary fields?

Reply:
Not sure of all summary fields, if you have some specific ones on you mind and have the energy for it, sure. I don't know how much work that would mean.

I think this method is maybe different from some average endpoint in the sense that validating the anchor is more important part of what it does. For some other endpoints, like e.g. add client, validation is more of a side effect. You don't call add client to validate the parameters, but you call /system/anchor/preview to validate the anchor (well, kind of).